example.py

```diff
@@ -198,6 +209,221 @@ def print_rpc_device(device: RpcDevice) -> None:
     print(f"Connected: {device.connected}")
 
 
+def save_endpoints(device: BlockDevice | RpcDevice) -> None:
+    """Save device endpoints."""
+    data_raw = {"shelly": device.shelly, "status": device.status}
+
+    if device.gen in BLOCK_GENERATIONS:
+        data_raw.update({"settings": device.settings})
```

Comment:
```suggestion
        data_raw.update({"settings": device.settings.copy()})
```

Since we are going to mutate this we should make a copy first